### ISSUE TITLE
Neural Networking output same value for different inputs

### ISSUE THREAD

**Issue by jthrelfall777** [beniz/deepdetect#136]
Hi,

The neural network seems to be outputting the same value for every input that is presented to the network. I've double checked my setup but can't seem to find what is wrong. This is happening on both docker images and compiled from source. It also doesn't seem to matter if it's CPU or GPU.

I trained everything from 40 iterations to 2000000 still the same. Please could I have some help?

https://gist.github.com/jthrelfall777/93064bc9c1fb7e63a0659e92d45e2651

Kind Regards,

John.

**Comment by beniz** [comment]
I will check, in the meantime, I recommend you do check the same dummy but with discrete classification instead of regression (e.g. replace labels with 0 or 1, and set regression to false), in order to rule out any external issue outside the regression setup.

**Comment by jthrelfall777** [comment]
Thanks, I will have a go with your suggestions.

**Comment by jthrelfall777** [comment]
I seem to be getting the same problem. The probability for all classes is the same and doesn't change.

**Comment by beniz** [comment]
I took a quick look, and you're almost there but there are many issues:

- your dataset is too small, you won't get far
- lower your `base_lr` to 0.0000000001
- remove the `min_vals`, `max_vals`, they are automatically computed by the server
- don't add `measure:f1` to the `/predict` call since your test data doesn't include label
- the `output` object in your `/predict` call is missplaced and thus ignored
- add some more layers, like `layers:[50,50]'
- set your `batch_size` to 5
- consider using `async:false` for small jobs and debugging like this.

I'd recommend you test on a toy problem with enough data, in the case above your net is not learning anything, and you can see that by looking at the `train_loss` parameter in training output.

**Comment by jthrelfall777** [comment]
Just an update I am still testing, I've had better results from the neural network now. There may still be an issue with scaling when testing the network. I seem to have to manually specify the values or I get an error message saying it couldn't load/find the values.

Does the network/software output only the scaled values back from the neural net? Would I have to convert these back to non scaled values manually?

**Comment by beniz** [comment]
Training computes the scaling parameters automatically and returns them when the call has completed. You then need to specify these values into subsequent predict calls. If you re-create the service after traning a model, you should be able to set these values within the `connector` object at service creation time, and this will record them for any subsequent predict call. Let me know if you have an issue.

Also, this page has an example, http://www.deepdetect.com/tutorials/csv-training/

**Comment by jthrelfall777** [comment]
Thanks for your help with this, I will let you know how it goes.
